05/DB.test.js

```diff
@@ -1 +1,106 @@
-import DB from './DB';
\ No newline at end of file
+import DB from "./DB";
+
+describe("DB", () => {
+  describe(".insert()", () => {
+    it("should reject when ID not a number", async () => {
+      const db = new DB();
+      const data = { id: "X" };
+      await expect(db.insert(data)).rejects.toMatch("ID can be only number!");
+    });
+
+    it("should reject when ID not unique", async () => {
+      const db = new DB();
+      const data = { id: 3, a: 2, b: 3 };
+      await db.insert(data);
+
+      await expect(db.insert(data)).rejects.toMatch("ID can't be duplicated!");
+    });
+
+    it("inserts data when no mistakes", async () => {
+      const db = new DB();
+      const data = { a: 2, b: 3 };
+
+      await expect(db.insert(data)).resolves.toEqual(data);
+    });
+  });
+
+  describe(".select()", () => {
+    it("selects data by ID", async () => {
+      const db = new DB();
+      const data = { id: 3, a: 2, b: 3 };
+      await db.insert(data);
+
+      await expect(db.select(data.id)).resolves.toEqual(data);
+    });
+
+    it("rejects when no matching ID", async () => {
```

Comment:
Warto dla pewności dawać na początku informację jakiej ilości  "testów/porównań/asercji" się spodziewamy tj. `expect.assertions(1);`

Ten test przechodzi, ale jak dodasz wspomniany kod to już nie. Mamy błąd:
![image](https://github.com/devmentor-pl/practice-js-testing/assets/55095257/d0654541-4fa1-4d17-b2d0-9d723a9d4a6f)


Okazuje się, że zapomniałeś o `await/return` przed `expect(...` .
Jeśli test nie ma błędu w asercji (a tak jest kiedy nie zostanie wykonanie żadne porównanie) to jest ok.
Tutaj kod testu wykona się szybciej niż funkcja asynchroniczna więc do porównania nie dochodzi ;)

Trzeba bardzo na to uważać :)